### DISCUSSION TITLE
Update incrementing logic to as to preserve filenames already ending with a number

### DISCUSSION THREAD

**Discussion by beesinblenders** [chhoumann/quickadd#698] [Ideas]
Right now, when a template choice is set to "increment" in case the file already exists, it creates a new file with the same name followed by a number. But when the original filename already ends with a number, it increments that number. For example if it's a journal entry like `2024-06-14.md`, the new file will be called `2024-06-15.md`. This leads to some confusion. 

Shouldn't it rather be something like `2024-06-14-1.md`?

**Comment by chhoumann** [comment]
Coming back to this one late - sorry about that, and good call.

This shipped in QuickAdd 2.12.1. The "file already exists" handling now has two separate create-new modes:

- **Append duplicate suffix**: `2024-06-14.md` -> `2024-06-14 (1).md`. This is the default when creating another file.
- **Increment trailing number**: changes trailing digits only (`Draft009.md` -> `Draft010.md`), so the date-bumping behavior you hit is now opt-in and labeled for what it actually does.

Pick "Append duplicate suffix" in your template choice's file-exists setting and journal entries stop getting bumped to the next day.

Docs: https://quickadd.obsidian.guide/docs/Choices/TemplateChoice

Closing as resolved.
